src/gas/determineGasFeeSuggestions.ts

```diff
@@ -91,20 +96,22 @@ export default async function determineGasFeeSuggestions({
         estimatedGasFeeTimeBounds: {},
         gasEstimateType: GAS_ESTIMATE_TYPES.LEGACY,
       };
-    }
-    throw new Error('Main gas fee/price estimation failed. Use fallback');
-  } catch {
```

Comment:
This code being replaced is an example of poor error handling, because we don't inspect what the error is before reacting to it. The author is assuming the error is operational (e.g. network not supported, connectivity issues, etc.), but this could in fact be hiding a mistake in our code. Because the error is swallowed indiscriminately, it hides any potential mistakes we may have made here, which will in turn make debugging such an error incredibly difficult.

The new utility function you've introduced seems to further entrench this pattern.